packages/schema/dtslint/Schema.ts

```diff
@@ -704,7 +704,7 @@ S.Struct({
 })
 
 // $ExpectType Struct<{ a: PropertySignature<":", "a" | "b", never, "?:", "a" | "b", true, never>; }>
-S.Struct({ a: S.optional(S.Literal("a", "b"), { default: () => "a", exact: true }) })
+S.Struct({ a: S.optional(S.Literal("a", "b"), { default: () => "a" as const, exact: true }) })
```

Comment:
Not completely sure why it was needed but as it was done for the other overload, I thought it was fine to add `as const`

Reply:
I'd revert this change, it's good to know that `as const` is not needed

Reply:
It was actually needed with my change. If I revert it, I have an error: 
![image](https://github.com/Effect-TS/effect/assets/15926755/746686a2-0679-49f7-b2e3-522f445f120e)
Any idea where this could come from?
I don't have the issue if I only extract `OptionalOptions`. It comes when I extract OptionalReturnType.

Reply:
ah, sorry, I read it wrong. I thought it was done for consistency, not because it was causing a new error. ok, let me see what could be causing it...

Reply:
I tries a few `in` or `out` in the `OptionalOptions` type but without success

Reply:
The issue seems related to `default` being a lazy arg `() => A`
I have an idea to solve the issue, but it involves quite a refactoring. I'll send another PR to replace this one